@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -1004,31 +1014,29 @@ export class TextElementsRenderer {
 
             catalog.showReplacementGlyphs = this.showReplacementGlyphs;
 
-            this.m_textRenderers.push({
-                fontCatalog: name,
-                textCanvas: loadedTextCanvas,
-                poiRenderer: this.m_poiRendererFactory.createPoiRenderer(loadedTextCanvas)
-            });
+            this.m_textCanvases.push(loadedTextCanvas);
         };
 
-        return this.m_fontCatalogLoader.loadCatalogs(catalogCallback).then(() => {
-            // Find the default TextCanvas and PoiRenderer.
-            let defaultTextCanvas: TextCanvas | undefined;
-            this.m_textRenderers.forEach(textRenderer => {
-                if (defaultTextCanvas === undefined) {
-                    defaultTextCanvas = textRenderer.textCanvas;
-                }
-            });
-            const defaultPoiRenderer = this.m_poiRendererFactory.createPoiRenderer(
-                defaultTextCanvas!
-            );
+        return this.m_fontCatalogLoader
+            .loadCatalogs(catalogCallback)
+            .then(() => {
+                // Find the default TextCanvas and PoiRenderer.
+                let defaultTextCanvas: TextCanvas | undefined;
+                this.m_textCanvases.forEach(textCanvas => {
+                    if (!defaultTextCanvas && textCanvas) {
+                        defaultTextCanvas = textCanvas;
+                    }
+                });
 
-            this.m_textStyleCache.initializeTextElementStyles(
-                defaultPoiRenderer,
-                defaultTextCanvas!,
-                this.m_textRenderers
-            );
-        });
+                this.m_textStyleCache.initializeTextElementStyles(
+                    this.m_poiRenderer!,
+                    defaultTextCanvas!,
```

Comment:
defaultTextCanvas will be undefined if there's no font catalog.

Reply:
done